File: Codigo/dindin-api/dindin-app/app/models/Goal.js
const { Model, DataTypes } = require("sequelize");

class Goal extends Model {
    static init(sequelize) {
        super.init(
            {
                id: {
                    type: DataTypes.INTEGER.UNSIGNED,
                    required: true,
                    primaryKey: true,
                    autoIncrement: true,
                    notEmpty: true,
                    allowNull: false,
                },
                wallet_id: {
                    type: DataTypes.INTEGER.UNSIGNED,
                    allowNull: true, // ! to change
                    // ! references: {
                    // !  model: Wallet,
                    // !  key: "id"
                    // !}
                },
                description: {
                    type: DataTypes.STRING(30),
                    allowNull: false,
                },
                value: {
                    type: DataTypes.DOUBLE,
                    allowNull: false,
                },
                status: {
                    type: DataTypes.ENUM,
                    values: ["FINISHED", "LOST", "PENDING"],
                    defaultValue: "PENDING",
                    allowNull: false,
                },
                type: {
                    type: DataTypes.ENUM,
                    values: ["A", "B"],
                    allowNull: false,
                    defaultValue: "A",
                },
                expire_at: {
                    allowNull: true,
                    type: DataTypes.DATE,
                },
                created_at: {
                    type: DataTypes.DATE,
                    allowNull: false,
                },
                updated_at: {
                    type: DataTypes.DATE,
                    allowNull: false,
                },
                deleted_at: {
                    type: DataTypes.DATE,
                    allowNull: true,
                    defaultValue: null,
                },
            },
            {
                tableName: "goal",
                charset: "utf8mb4",
                collate: "utf8mb4_bin",
                timestamps: true, // deleted_at and updatedAt need this
                paranoid: true, // deleted_at need this
                createdAt: "created_at",
                updatedAt: "updated_at",
                deletedAt: "deleted_at", // .destroy() and .destroy(); to softdelete
                sequelize,
            }
        );
    }
}

module.exports = Goal;

File: Codigo/dindin-mobile/dindin/lib/models/goal.dart
class Goal {
  final num id;
  final String description;
  final num? value;
  final String? status;
  final String? type;
  final String? expireAt;
  final num? walletId;
  final String? createdAt;
  final String? updatedAt;
  final String? deletedAt;

  const Goal({
    required this.id,
    required this.description,
    required this.value,
    required this.status,
    required this.type,
    required this.expireAt,
    required this.walletId,
    required this.createdAt,
    required this.updatedAt,
    required this.deletedAt,
  });

  factory Goal.fromJson(Map<String, dynamic> json) {
    return Goal(
      id: int.parse(json['id']),
      description: json['description'],
      value: json['value'],
      status: json['status'],
      type: json['type'],
      expireAt: json['expire_at'],
      walletId: int.parse(json['wallet_id']),
      createdAt: json['created_at'],
      updatedAt: json['updated_at'],
      deletedAt: json['deleted_at'],
    );
  }
}
